# Copied Notebook from bassam and did some changes by myself

In [18]:
from sklearn.preprocessing import LabelEncoder
import pandas  as pd
data=pd.read_csv("../Data/raw/transactions.csv")
print("Data shape:", data.shape)

Data shape: (199999, 10)


In [19]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 199999 entries, 0 to 199998
Data columns (total 10 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   step            199999 non-null  int64  
 1   type            199999 non-null  object 
 2   amount          199999 non-null  float64
 3   nameOrig        199999 non-null  object 
 4   oldbalanceOrg   199999 non-null  float64
 5   newbalanceOrig  199999 non-null  float64
 6   nameDest        199999 non-null  object 
 7   oldbalanceDest  199999 non-null  float64
 8   newbalanceDest  199999 non-null  float64
 9   isFraud         199999 non-null  int64  
dtypes: float64(5), int64(2), object(3)
memory usage: 15.3+ MB


In [20]:
# Check for duplicates
data.duplicated().sum()

0

In [21]:
# Check for balance
data['isFraud'].value_counts(normalize=True)

0    0.99859
1    0.00141
Name: isFraud, dtype: float64

In [22]:
data['isFraud'].value_counts()

0    199717
1       282
Name: isFraud, dtype: int64

In [23]:
data.columns

Index(['step', 'type', 'amount', 'nameOrig', 'oldbalanceOrg', 'newbalanceOrig',
       'nameDest', 'oldbalanceDest', 'newbalanceDest', 'isFraud'],
      dtype='object')

In [6]:
data.head(5)

step      type     amount     nameOrig  oldbalanceOrg  newbalanceOrig  \
0     8  CASH_OUT  158007.12   C424875646           0.00            0.00   
1   236  CASH_OUT  457948.30  C1342616552           0.00            0.00   
2    37   CASH_IN  153602.99   C900876541    11160428.67     11314031.67   
3   331  CASH_OUT   49555.14   C177696810       10865.00            0.00   
4   250  CASH_OUT   29648.02   C788941490           0.00            0.00   

      nameDest  oldbalanceDest  newbalanceDest  isFraud  
0  C1298177219       474016.32      1618631.97        0  
1  C1323169990      2720411.37      3178359.67        0  
2   C608741097      3274930.56      3121327.56        0  
3   C462716348            0.00        49555.14        0  
4  C1971700992        56933.09        86581.10        0

In [7]:
print(data.nameOrig.unique().shape)
print(data.nameDest.unique().shape)

(199990,)
(174523,)


### Conclusion
- `isFraud` Classes are very imbalanced - (99.9% to 0.0001%)
- No `null/missing` values
- Too many unique values

- "clear imbalance classes case here. Random Forest will take care of this shit, ion get paid enough for this nonsense" - Bassam

In [9]:
# Drop Irrelevant columns
data.drop(columns=['nameOrig', 'nameDest'], inplace=True)

In [10]:
le = LabelEncoder()

data['type'] = le.fit_transform(data['type'])

In [11]:
data.head()

step  type     amount  oldbalanceOrg  newbalanceOrig  oldbalanceDest  \
0     8     1  158007.12           0.00            0.00       474016.32   
1   236     1  457948.30           0.00            0.00      2720411.37   
2    37     0  153602.99    11160428.67     11314031.67      3274930.56   
3   331     1   49555.14       10865.00            0.00            0.00   
4   250     1   29648.02           0.00            0.00        56933.09   

   newbalanceDest  isFraud  
0      1618631.97        0  
1      3178359.67        0  
2      3121327.56        0  
3        49555.14        0  
4        86581.10        0

In [12]:
data.to_csv("../Data/processed/processed.csv", index=False)